interface/patient_file/pos_checkout_normal.php

```diff
@@ -0,0 +1,1158 @@
+              );
+              sqlCommitTrans();
+        }
+
+      // If applicable, set the invoice reference number.
+        $invoice_refno = '';
+        if (isset($_POST['form_irnumber'])) {
+            $invoice_refno = trim($_POST['form_irnumber']);
+        } else {
+            $invoice_refno = BillingUtilities::updateInvoiceRefNumber();
+        }
+        if ($invoice_refno) {
+            sqlStatement("UPDATE form_encounter " .
+            "SET invoice_refno = ? " .
+            "WHERE pid = ? AND encounter = ?", array($invoice_refno,$form_pid,$form_encounter));
+        }
+
+        generate_receipt($form_pid, $form_encounter);
+        exit();
+    }
+
+    // If an encounter ID was given, then we must generate a receipt.
+    //
+    if (!empty($_GET['enc'])) {
+        generate_receipt($patient_id, $_GET['enc']);
+        exit();
+    }
+
+    // Get the unbilled billing table items for this patient.
+    $query = "SELECT id, date, code_type, code, modifier, code_text, " .
+      "provider_id, payer_id, units, fee, encounter " .
+      "FROM billing WHERE pid = ? AND activity = 1 AND " .
+      "billed = 0 AND code_type != 'TAX' " .
+      "ORDER BY encounter DESC, id ASC";
+    $bres = sqlStatement($query, array($patient_id));
+
+    // Get the product sales for this patient.
+    $query = "SELECT s.sale_id, s.sale_date, s.prescription_id, s.fee, " .
+      "s.quantity, s.encounter, s.drug_id, d.name, r.provider_id " .
+      "FROM drug_sales AS s " .
+      "LEFT JOIN drugs AS d ON d.drug_id = s.drug_id " .
+      "LEFT OUTER JOIN prescriptions AS r ON r.id = s.prescription_id " .
+      "WHERE s.pid = ? AND s.billed = 0 " .
+      "ORDER BY s.encounter DESC, s.sale_id ASC";
+    $dres = sqlStatement($query, array($patient_id));
+
+    // If there are none, just redisplay the last receipt and exit.
+    //
+    if (sqlNumRows($bres) == 0 && sqlNumRows($dres) == 0) {
+        generate_receipt($patient_id);
+        exit();
+    }
+
+    // Get the valid practitioners, including those not active.
+    $arr_users = array();
+    $ures = sqlStatement("SELECT id, username FROM users WHERE " .
+      "( authorized = 1 OR info LIKE '%provider%' ) AND username != ''");
+    while ($urow = sqlFetchArray($ures)) {
+        $arr_users[$urow['id']] = '1';
+    }
+
+    // Now write a data entry form:
+    // List unbilled billing items (cpt, hcpcs, copays) for the patient.
+    // List unbilled product sales for the patient.
+    // Present an editable dollar amount for each line item, a total
+    // which is also the default value of the input payment amount,
+    // and OK and Cancel buttons.
+    ?>
+<!DOCTYPE html>
+    <html>
+    <head>
+        <?php Header::setupHeader(['datetime-picker']);?>
+
+        <script>
+            var mypcc = <?php echo js_escape($GLOBALS['phone_country_code']); ?>;
+
+            <?php require($GLOBALS['srcdir'] . "/restoreSession.php"); ?>
```

Comment:
same question as above (ie. why are popups including this)?
(note this is not really part of this code review since existing code; just trying to figure out why some of the popups are doing this)

Reply:
Sometimes top is not available or in scope. I'm thinking we could add a dependency check in utility then include if needed.

Reply:
Yeah. Guess I'll leave this alone for now, but at some point it should be changed and tested.